dvc/repo/metrics/show.py

```diff
@@ -188,9 +188,15 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
         if not o.metric:
             continue
 
-        # NOTE this case assumes that metric has been modified with
-        # `dvc metric modify -t {} -x {}`, so stage file should contain
-        # info what type metric is and what is its xpath
+        # NOTE this case assumes:
+        # - that metric has been modified with `dvc metric modify -t {} -x {
```

Comment:
it's not that metric is modified, it's more like "there are type and path defined in the DVC-file for this metric". There are other way to modify the file, first of all. Second, I can modify it and remove the previous definition.

Reply:
I removed also xpath, typ is core thing here. Xpath is explained later